content/design-systems-for-developers/react/ja/architecture.md

```diff
@@ -0,0 +1,110 @@
+---
+title: 'システム設計'
+tocTitle: '設計'
+description: 'コンポーネントライブラリからデザインシステムを抽出する方法'
+commit: '798276b'
+---
+
+第 2 章では、既存のコンポーネントライブラリからデザインシステムを抽出します。その過程で、どのコンポーネントがデザインに属するのか決定し開発者が着手時に直面する共通の課題を浮き彫りにします。
+
+大企業では、この活動はデザイン、エンジニアリング、そしてプロダクトチームの協働で行われます。[Chromatic](https://www.chromatic.com/) (Storybook の開発会社)と[Storybook](https://storybook.js.org/)は 3 つ以上の領域に渡り 1400 超のオープンソースの貢献者を擁する活発なフロントエンド基盤を共有しているため、あなたのために当プロセスの概略を伝えていきます。
+
+## 課題
+
+あなたが開発チームで働いているなら、おそらく大きなチームはあまり効率的ではないことに気付いているでしょう。チームが大きくなるにしたがいコミュニケーションロスがはびこります。既存の UI パターンが文書化されないか完全に失われます。それは開発者が新しい機能を開発する代わりに車輪を再生産することを意味します。時が経つにつれ、プロジェクトは 1 回限りのコンポーネントで山積みになります。
+
+私たちはこの困難な状況へ飛び込みました。経験を積んだチームの望ましい意志にも関わらず、UI コンポーネントは終わることなく再開発されるかコピー＆ペーストされていました。UI パターンは外観と機能性の点で同じ源流だと思われていました。各コンポーネントは新参の開発者にとって信頼できる情報源と見なすのは不可能でしたし、まして貢献など無理なものでした。
+
+![UIs diverge](/design-systems-for-developers/design-system-inconsistent-buttons.jpg)
+
+## デザインシステムを作成する
+
+デザインシステムはパッケージマネージャ経由で中心的によくメンテナンスされたリポジトリにおいて共通の UI コンポーネントを強固なものにします。開発者は複数のプロジェクト間で同じ UI コードをペーストする代わりに標準化された UI コンポーネントをインポートします。
+
+多くのデザインシステムはゼロから構築されていません。代わりに、いち企業においてデザインシステムとして再パッケージされて用いられる検証済みの UI コンポーネントから作成されます。私たちのプロジェクトも例外ではありません。時間の節約のため既存の本番コンポーネントライブラリからいいとこ取りをしてデザインシステムをステークホルダーへいち早く届けます。
+
+![What's in a design system](/design-systems-for-developers/design-system-contents.jpg)
+
+## デザインシステムはどこにあるのですか？
+
+デザインシステムは別のコンポーネントライブラリだと考えることができます、しかしそれはいちアプリだけではなく組織全体に提供しているのです。デザインシステムは UI の基礎にフォーカスする一方で、特定プロジェクトのコンポーネントライブラリは複合コンポーネントからスクリーンまであらゆる箇所に含まれます。
+
+そういうわけで、デザインシステムはいかなるプロジェクトからも独立しかつ全てのプロジェクトが従属する必要があります。変更はパッケージマネージャで配布されるバージョン管理されたパッケージ経由で組織全体に渡り伝播します。プロジェクトはデザインシステムのコンポーネントを再利用し更に必要であればカスタマイズできます。これらの制約はフロントエンドのプロジェクトを構成する青写真を提供します。
+
+![Who uses a design system](/design-systems-for-developers/design-system-consumers.jpg)
+
+## create-react-app と GitHub でリポジトリをセットアップする
+
+React は[State of JS](https://stateofjs.com/)の調査によると最も人気のあるビューレイヤーです。Storybook の大多数が React を使っているので、それを[create-react-app](https://github.com/facebook/create-react-app)のボイラープレートにそって使います。
+
+コマンドラインで、次のコマンドを実行してください:
+
+```shell
+# ファイルをクローン
+npx degit chromaui/learnstorybook-design-system-template learnstorybook-design-system
+
+cd learnstorybook-design-system
+
+# 依存関係をインストール
+yarn install
```

Comment:
@naoki8ster one item I need to follow up with you. I've checked and it seems that you have the comments inside the code snippets translated as well, despite understanding that it provides a clearer picture to the reader, we always recommend those comments inside the snippets to be in English. If you could revert them back we'd appreciate it.

Reply:
@jonniebigodes Thanks for your review! I've just reverted some files which includes Japanese comments inside the snippets.

Reply:
@naoki8ster no need to thank whatsoever. I'm the one that should be thanking you for addressing the feedback so promptly! Appreciate it 🙏 !